tests/trestle/core/remote/cache_test.py

```diff
@@ -0,0 +1,371 @@
+        with pytest.raises(err.TrestleError):
+            fetcher.get_oscal(Catalog)
+        oscal_read_mock.assert_called_once()
+
+
+def test_fetcher_base(tmp_trestle_dir):
+    """Test whether fetcher can get an object from the cache."""
+    # Fetch from local content, expecting it to be cached and then fetched.
+    rand_str = ''.join(random.choice(string.ascii_letters) for x in range(16))
+    catalog_file = pathlib.Path(tmp_trestle_dir / f'{rand_str}.json').__str__()
+    catalog_data = generators.generate_sample_model(Catalog)
+    catalog_data.oscal_write(pathlib.Path(catalog_file))
+    saved_data = fs.load_file(pathlib.Path(catalog_file))
+    fetcher = cache.FetcherFactory.get_fetcher(pathlib.Path(tmp_trestle_dir), catalog_file, False, False)
+    # Create/update the cache copy
+    fetcher._refresh = True
+    fetcher._cache_only = False
+    fetched_data = fetcher.get_raw()
+    assert fetched_data == saved_data
+
+
+def test_github_fetcher():
+    """Test the github fetcher."""
+    pass
+
+
+def test_local_fetcher_get_fails(tmp_trestle_dir):
+    """Test the local fetcher."""
+    rand_str = ''.join(random.choice(string.ascii_letters) for x in range(16))
+    catalog_file = pathlib.Path(tmp_trestle_dir / f'{rand_str}.json').__str__()
+    catalog_data = generators.generate_sample_model(Catalog)
+    catalog_data.oscal_write(pathlib.Path(catalog_file))
+    fetcher = cache.FetcherFactory.get_fetcher(pathlib.Path(tmp_trestle_dir), catalog_file, False, False)
+    fetcher._cache_only = True
+    with pytest.raises(err.TrestleError):
+        fetcher.get_raw()
+    with pytest.raises(err.TrestleError):
+        fetcher.get_oscal(Catalog)
+
+
+def test_local_fetcher(tmp_trestle_dir):
+    """Test the local fetcher."""
+    rand_str = ''.join(random.choice(string.ascii_letters) for x in range(16))
+    catalog_file = pathlib.Path(tmp_trestle_dir / f'{rand_str}.json').__str__()
+    catalog_data = generators.generate_sample_model(Catalog)
+    catalog_data.oscal_write(pathlib.Path(catalog_file))
+    fetcher = cache.FetcherFactory.get_fetcher(pathlib.Path(tmp_trestle_dir), catalog_file, False, False)
+    fetcher._refresh = True
+    fetcher._cache_only = False
+    fetcher._update_cache()
+    assert fetcher._inst_cache_path.exists()
+
+
+def test_https_fetcher(tmp_trestle_dir):
+    """Test the https fetcher."""
+    os.environ['fetcher_user'] = 'fetcher'
+    os.environ['fetcher_pass'] = 'secret'
+    # This is a real, live URL:
+    uri = 'https://{{fetcher_user}}:{{fetcher_pass}}@placekitten.com/200/300'
+    fetcher = cache.FetcherFactory.get_fetcher(pathlib.Path(tmp_trestle_dir), uri, False, False)
+    fetcher._refresh = True
+    fetcher._cache_only = False
+    with patch('requests.models.Response.json') as json_mock:
+        json_mock.return_value = {'isBinary': False, 'text': str({'key': 'val'})}
+        try:
+            fetcher._update_cache()
+        except Exception:
+            AssertionError()
```

Comment:
Why is the try-except block needed here?

Reply:
Code-wise, the _update_cache() could throw exceptions, although you're right, it's not expected here assuming that the given live url isn't broken. I've updated this to instead do a proper test for the fetched object. We can subsequently update such tests with a local https service instead.